plugins/module_utils/client.py

```diff
@@ -44,6 +44,12 @@ def __init__(
         refresh_token=None, client_id=None, client_secret=None, timeout=None
     ):
         self.host = host
+        if not self.host:
+            raise AuthError("Instance host cannot be None.")
+
+        if not self.host.startswith(('https://', 'http://')):
+            raise AuthError("Invalid instance host value. Value must start with 'https://' or 'http://'")
```

Comment:
These two errors are not authentication errors (we have nothing valid to authenticate against). So we should be raising a plain `ServiceNowError` instead. We can also simplify this like this:

```suggestion
        if not (host or "").startswith(('https://', 'http://')):
            raise AuthError("Invalid instance host value: '{0}'. Value must start with 'https://' or 'http://'".format(host))
```